tests/jax/components/building/networks_test.py

```diff
@@ -0,0 +1,212 @@
+        networks_config.seed = 919
+
+        super().__init__(networks_config)
+
+
+@pytest.fixture
+def test_network_factory() -> Callable:
+    """Pytest fixture for network factory.
+
+    Returns:
+        Network factory using custom make_default_networks.
+    """
+
+    def make_default_networks(
+        environment_spec: MAEnvironmentSpec,
+        agent_net_keys: Dict[str, str],
+        rng_key: List[int],
+        net_spec_keys: Dict[str, str] = {},
+        policy_layer_sizes: Sequence[int] = (
+            256,
+            256,
+            256,
+        ),
+        critic_layer_sizes: Sequence[int] = (512, 512, 256),
+    ) -> Dict[str, Any]:
+        net_keys = {"net_1", "net_2", "net_3"}
+        networks = {}
+
+        for net_key in net_keys:
+            networks[net_key] = {
+                "environment_spec": environment_spec,
+                "agent_net_keys": agent_net_keys,
+                "rng_key": rng_key,
+                "net_spec_keys": net_spec_keys,
+                "policy_layer_sizes": policy_layer_sizes,
+                "critic_layer_sizes": critic_layer_sizes,
+                "net_key": net_key,
+            }
+
+        return networks
+
+    def network_factory(*args: Any, **kwargs: Any) -> Any:
+        return make_default_networks(  # type: ignore
+            policy_layer_sizes=(254, 254, 254),
+            critic_layer_sizes=(512, 512, 256),
+            *args,
+            **kwargs,
+        )
+
+    return network_factory
+
+
+@pytest.fixture
+def test_builder() -> SystemBuilder:
+    """Pytest fixture for system builder.
+
+    Adds mock env specs and agent net keys to store.
+
+    Returns:
+        System builder with no components.
+    """
+    test_builder = Builder(components=[])
+    test_builder.store.environment_spec = make_fake_env_specs()
+    test_builder.store.agent_net_keys = {"key1": "value1", "key2": "value2"}
+    return test_builder
+
+
+@pytest.fixture
+def test_default_networks(test_network_factory: Callable) -> Networks:
+    """Pytest fixture for default networks.
+
+    Args:
+        test_network_factory: factory to use in default networks config.
+
+    Returns:
+        Default networks test component.
+    """
+    return TestDefaultNetworks(test_network_factory)
+
+
+def test_key_in_store(
+    test_default_networks: Networks, test_builder: SystemBuilder
+) -> None:
+    """Test if key is loaded into the store by build.
+
+    Args:
+        test_default_networks: Pytest fixture for default networks component.
+        test_builder: Pytest fixture for test system builder
+
+    Returns:
+        None.
+    """
+    assert not hasattr(test_builder.store, "key")
+    test_default_networks.on_building_init_start(test_builder)
+    assert hasattr(test_builder.store, "key")
+    assert isinstance(test_builder.store.key, jax.random.PRNGKeyArray) or isinstance(
+        test_builder.store.key, jax.numpy.DeviceArray
```

Comment:
Just checking about the or here. Does the `PRNGKeyArray` change to a `DeviceArray` later on?

Reply:
There is some weird `jax` behaviour here. The function `jax.random.PRNGKey(self.config.seed)` is supposed to produce a `PRNGKeyArray` according to its type signature, but it doesn't. In practice, it returns a `DeviceArray`.

To make it worse, elsewhere it says it returns a `KeyArray` which is just `KeyArray = Union[Array, prng.PRNGKeyArray]` which is just `Array = Any`. So it's just `Any`, which is useless.

My solution felt the most robust to me. I could just change them to `jax.numpy.DeviceArray` instead?

Reply:
Ah, thank you for clarifying. I think your solution is the best in that case.